python/ngen_cal/src/ngen/cal/ngen.py

```diff
@@ -361,9 +415,17 @@ def __init__(self, **kwargs):
             #FIXME pick up params per catchmment somehow???
             params = _map_params_to_realization(self.params, catchment)
             catchments.append(AdjustableCatchment(self.workdir, id, nexus, params))
+        
+        if self.eval_feature:
+            for n in eval_nexus:
+                wb = self._flowpath_hydro_fabric[ self._flowpath_hydro_fabric['toid'] == n.id ]
```

Comment:
Little code golf for you.
```suggestion
                key = self._flowpath_hydro_fabric.loc[ self._flowpath_hydro_fabric['toid'] == n.id, 'name'].head(1)
```

Reply:
This is the hazards of complex indexing -- it can be written many ways, and miss understood in all of them haha.  I'm in favor of scoring this golf round by readability and complexity. 

Also, I'm not convinced that this will actually work, because `name` is *not* a column in the dataframe -- the `name` attribute used becomes essentially the index value of the returned series from `iloc`.